[DOCS] Adds certutil http command to TLS setup steps (#51241)
elastic/elasticsearch: x-pack/docs/en/security/securing-communications/tls-http.asciidoc

This command generates a zip file that contains certificates and keys for use in {es} and {kib}. Each folder contains a readme that explains how to use the files.
--

. Verify that you've copied the output files to the appropriate locations, as
specified in the readme files. 
+
--
For example, copy the `http.p12` file from the `elasticsearch` folder into a 
directory within the {es} configuration directory on each node. If you chose to 
generate one certificate per node, copy the appropriate `http.p12` file to each
node. If you want to use {kib} to access this cluster, copy the
`elasticsearch-ca.pem` file from the `kibana` folder into the {kib}
configuration directory.
--

. Enable TLS and specify the information required to access the node’s
certificate. Follow the instructions in the readme in the `elasticsearch` folder
of the zip file. For example:

..Add the following information to the `elasticsearch.yml` file on each node:
+
--
[source, yaml]
--------------------------------------------------
xpack.security.http.ssl.enabled: true
xpack.security.http.ssl.keystore.path: "http.p12"
--------------------------------------------------

> If you did not use the `elasticsearch-certutil http` command and you have

Reviewer: Using `elasticsearch-certutil http` doesn't necessarily mean they did not end up with a PEM file. For instance, they could have used `elasticsearch-certutil http` to generate a CSR that they then took to their CA to sign and get back a PEM encoded certificate

bin/elasticsearch-certutil http
----------------------------------------------------------

This command generates a zip file that contains certificates and keys for use in {es} and {kib}. Each folder contains a readme that explains how to use the files.
--

. Verify that you've copied the output files to the appropriate locations, as
specified in the readme files. 
+
--
For example, copy the `http.p12` file from the `elasticsearch` folder into a 
directory within the {es} configuration directory on each node. If you chose to 
generate one certificate per node, copy the appropriate `http.p12` file to each
node. If you want to use {kib} to access this cluster, copy the
`elasticsearch-ca.pem` file from the `kibana` folder into the {kib}
configuration directory.
--

. Enable TLS and specify the information required to access the node’s
certificate. Follow the instructions in the readme in the `elasticsearch` folder
of the zip file. For example:

.. Update the `elasticsearch.yml` file on each node with the location of the
certificates.
+
--
If the certificates are in PKCS#12 format:

[source, yaml]
--------------------------------------------------

> xpack.security.http.ssl.enabled: true

Author: @tvernum I noticed that the output from the elasticsearch-certutil http command did not mention setting the truststore.path, so I've removed it here too.  If it's still required, let me know and I'll re-add it.

Reviewer 2: It's not required.

Technically, setting a truststore would  configure which certificates the server would trust if we had clientAuthentication enabled (PKI). We don't need that.

It's also used by setup-passwords, but if no truststore is configured, it will assume that the keystore has the correct certificates and trust that, so it all works without needing any special config.

Author: Great, thanks!

> xpack.security.http.ssl.keystore.path: "http.p12"

Author: I think that since we talked about using a certs folder in previous examples, it should be used here too:
```suggestion
xpack.security.http.ssl.keystore.path: certs/http.p12
```

Reviewer 2: Maybe we should drop the `certs/` part entirely? 
We don't use it in the PEM examples (and that uses more files)

{es} and {kib}. Each folder contains a readme that explains how to use the files.
--

. Verify that you've copied the output files to the appropriate locations, as
specified in the readme files. 
+
--
For example, copy the `http.p12` file from the `elasticsearch` folder into a 
directory within the {es} configuration directory on each node. If you chose to 
generate one certificate per node, copy the appropriate `http.p12` file to each
node. If you want to use {kib} to access this cluster, copy the
`elasticsearch-ca.pem` file from the `kibana` folder into the {kib}
configuration directory.
--

. Enable TLS and specify the information required to access the node’s
certificate. Follow the instructions in the readme in the `elasticsearch` folder
of the zip file. For example:

.. Update the `elasticsearch.yml` file on each node with the location of the
certificates.
+
--
If the certificates are in PKCS#12 format:

[source, yaml]
--------------------------------------------------
xpack.security.http.ssl.enabled: true
xpack.security.http.ssl.keystore.path: "http.p12"
--------------------------------------------------

> If the certificates are in PEM format:

Reviewer: I think this section might be slightly confusing now. We document with the assumption that `bin/elasticsearch-certutil http` is used and suggest accordingly but there is no possibility for the tool to ouput PEM files like `node.key` and `node.crt` that we use in this example. 
`node.crt` and `node.key` also match the examples we use in the transport ssl section above, which might add to the confusion. 

Maybe if we remove "Follow the instructions in the readme in the elasticsearch folder of the zip file. " from step 3, that will make the step more generic. If users have used the `http` subcommand, they have copied the files already ( as suggested in previous step ) , if not, they have their keys and certificates ready at hand. We can still use `http.p12` for consistency here and also use `node1_http.key` and `node1_http.crt` in the PEM example to denote that these are usually different than the ones used for transport TLS. WDYT ?

Reviewer 2: Since the tool supports generating CSRs, and outputs a PEM key for that, we need this section, but I agree the wording is a little awkward.
That said, people know what type of files they have (or at least, if they don't know then I don't think restructuring the content will really help), so I'm OK with it as is.

Something simple like
```suggestion
If you have certificates in PEM format:
```
would be a little clearer, but it's a tiny difference.

Reviewer: Ι'm fine with this also now that we changed the key/certificate file names here

    must be a location within the {es} configuration directory.
--

.. If you secured the keystore with a password, add that password to a secure

> configuration setting in {es}:

Reviewer: ```suggestion
setting in {es}:
```

I think we call them `secure settings` ( as opposed to `secure configuration settings`)  elsewhere

    must be a location within the {es} configuration directory.
--

> .. If you secured the certificate with a password, add that password to a secure 

Reviewer: ```suggestion
.. If you secured the keystore or the private key with a password, add that password to a secure 
```
